include/circt/Dialect/Verif/VerifOps.td

```diff
@@ -40,4 +42,32 @@ def CoverOp : AssertLikeOp<"cover"> {
   let summary = "Ensure that a property can hold.";
 }
 
+def FormatStringOp : VerifOp<"fstr", [
+    Pure
+  ]> {
+  let summary = "Creates a formatted string.";
+  let description = [{
+    Creates a formatted string suitable for printing via the `verif.printf` op.
+    The formatting syntax is (for now) expected to be identical to verilog
+    string formatting to keep things simple for emission. This should change
+    once this op is expected to be used for more than just emitting sv.
+  }];
+  let arguments = (ins
+    StrAttr:$formatString,
+    Variadic<AnyType>:$substitutions);
+  
+  let results = (outs HWStringType:$str);
+  let assemblyFormat = [{
+    $formatString `(` $substitutions `)` `:` type($substitutions) attr-dict
+  }];
+}
+
+def PrintfOp : VerifOp<"printf", []> {
+  let summary = "Prints a formatted message.";
+  let arguments = (ins HWStringType:$formattedString);
+  let assemblyFormat = [{
+    $formattedString attr-dict
+  }];
+}
```

Comment:
I think I would just call this `PrintOp` and `verif.print`, without the "f". This operation doesn't really do any formatting, it just prints a string. The formatting happens with the above op. It would be cool if the formatting didn't leak into the printing op!

We could argue that this being emitted as `sv.fwrite` is just an optimized lowering that's able to lower `PrintOp(FormatStringOp(...))` directly. But we could also just lower the `FormatStringOp` into an fully interpolated SV string first, and then do an unformatted print afterwards.